src/core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java

```diff
@@ -111,6 +110,33 @@ public abstract class KylinConfigBase implements Serializable {
 
     protected static final Map<String, String> STATIC_SYSTEM_ENV = new ConcurrentHashMap<>(System.getenv());
 
+    // It's a workaround to avoid lock in bottom hash table
+    // It can be removed after updating JDK to 11
+    protected static final ConcurrentHashMap<Object, Object> STATIC_SYSTEM_PROPERTY = new ConcurrentHashMap<>(
+            System.getProperties());
+
+    protected static String getSystemProperty(String key) {
+        Object oval = STATIC_SYSTEM_PROPERTY.get(key);
```

Comment:
system.getProperty() used the checkKey() for fast failed, so this place may check too.

Reply:
Have added checkKey() for getSystemProperty.